controllers/cluster_controller.go

```diff
@@ -615,9 +625,112 @@ func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&hwameistoroperatorv1alpha1.Cluster{}).
 		Owns(&appsv1.DaemonSet{}).
 		Owns(&appsv1.Deployment{}).
+		Owns(&batchv1.Job{}).Watches(&source.Kind{Type: &batchv1.Job{}},
+		&handler.EnqueueRequestForObject{},
+		builder.WithPredicates(
+			predicate.Funcs{
+				UpdateFunc: r.handleDrbdJobUpdatePredicate,
+				DeleteFunc: r.handleDrbdJobDeletePredicate,
+			},
+		)).
 		Complete(r)
 }
 
+func (r *ClusterReconciler) handleDrbdJobUpdatePredicate(e event.UpdateEvent) bool {
+	newJob := e.ObjectNew.(*batchv1.Job)
+	r.handleDrbdJobUpdate(r.Client, newJob)
+	return true
+}
+
+func (r *ClusterReconciler) handleDrbdJobDeletePredicate(e event.DeleteEvent) bool {
+	job := e.Object.(*batchv1.Job)
+	r.handleDrbdJobDelete(r.Client, job)
+	return false
+}
+
+func (r *ClusterReconciler) handleDrbdJobUpdate(cli client.Client, job *batchv1.Job) error {
+	if !isJobComplete(job) {
+		return nil
+	}
+	log.Infof("Job %s has successfully completed.", job.Name)
+	var nodeList corev1.NodeList
+	if err := cli.List(context.TODO(), &nodeList); err != nil {
+		log.Errorf("List nodes err: %v", err)
+		return err
+	}
+	var jobList batchv1.JobList
+	if err := cli.List(context.TODO(), &jobList, client.InNamespace("hwameistor")); err != nil {
+		log.Errorf("List jobs err: %v", err)
+		return err
+	}
+	count := 0
+	for _, item := range jobList.Items {
+		if strings.HasPrefix(item.Name, "drbd-adapter") && isJobComplete(&item) {
+			count++
+		}
+	}
+
+	if count >= len(nodeList.Items) {
+		instance := &hwameistoroperatorv1alpha1.Cluster{}
+		if err := cli.Get(context.TODO(), types.NamespacedName{Name: "hwameistor-cluster", Namespace: "hwameistor"}, instance); err != nil {
+			log.Errorf("Get cluster instance err: %v", err)
+			return err
+		}
+		cluster := instance.DeepCopy()
+		cluster.Status.DRBDAdapterCreated = true
+		cluster.Status.DRBDAdapterCreatedJobNum = count
+		if err := cli.Status().Update(context.TODO(), cluster); err != nil {
+			log.Errorf("Update status err: %v", err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *ClusterReconciler) handleDrbdJobDelete(cli client.Client, job *batchv1.Job) error {
+	instance := &hwameistoroperatorv1alpha1.Cluster{}
+	if err := cli.Get(context.TODO(), types.NamespacedName{Name: "hwameistor-cluster", Namespace: "hwameistor"}, instance); err != nil {
+		log.Errorf("Get cluster instance err: %v", err)
+		return err
+	}
+	if instance.Status.DRBDAdapterCreatedJobNum > 0 {
+		return nil
+	}
+
+	newJob := job.DeepCopy() // Create a copy of the deleted job
+	// Generate a unique name for the new Job
+	newJobName := fmt.Sprintf("%s-%d", job.Name, time.Now().Unix())
+	newJob.ObjectMeta = metav1.ObjectMeta{
+		Name:      newJobName, // Use a unique name
```

Comment:
this may cause invalid job name when jobs being deleted mutil times

Reply:
I modified this part, please take a look